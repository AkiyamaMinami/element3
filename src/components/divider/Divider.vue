<template>
  <div v-bind="$attrs" :class="['el-divider', `el-divider--${direction}`]">
    <div
      v-if="$slots.default && direction !== 'vertical'"
      :class="['el-divider__text', `is-${contentPosition}`]"
    >
      <slot />
    </div>
  </div>
</template>
<script setup lang="ts">
import {withDefaults} from 'vue'
interface DividerTyoe {
  direction?:'horizontal'|'vertical',
  contentPosition?:'left'|'center'|'right'
}

withDefaults(defineProps<DividerTyoe>(),{
  direction:'horizontal',
  contentPosition:'center'
})

</script>
<style lang="scss">
@import '../../theme/src/divider.scss'

</style>
