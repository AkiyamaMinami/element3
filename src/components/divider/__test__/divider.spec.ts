import { mount } from '@vue/test-utils'
import Divider from '../Divider.vue'

describe('Divider.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(Divider)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('content', () => {
    const vm = mount(Divider, {
      slots: {
        default: () => 'Im perfect divider!',
      },
    })

    expect(vm.text()).toBe('Im perfect divider!')
  })

  it('direction', () => {
    const vm = mount(Divider, {
      props: { direction: 'vertical' },
    })

    expect(vm.classes()).toContain('el-divider--vertical')
  })

  it('contentPosition', () => {
    const vm = mount(Divider, {
      props: { 'content-position': 'left' },
      slots: {
        default: () => {
          return 'some text'
        },
      },
    })
    const text = vm.find('.el-divider__text')

    expect(text).toBeDefined()
    expect(text.classes()).toContain('is-left')
  })

  it('dividerStyle', () => {
    const vm = mount(Divider, {
      props: { dividerStyle : 'double' }
    })

    expect(vm.classes()).toContain(`is-double`)
  })

  it('color', () => {
    const vm = mount(Divider, {
      props: { color: 'rgb(23, 45, 52)' }
    })

    expect(vm.attributes('style')).toBe('color: rgb(23, 45, 52);')
  })

  it('dark', async () => {
    const vm = mount(Divider, {
      slots: {
        default: 'Divider'
      },
      props: { dark: true }
    })

    expect(vm.find('.el-divider__text').classes()).toContain('is-dark')

    await vm.setProps({
       dark: false
    })

    expect(vm.find('.el-divider__text').classes()).not.toContain('is-dark')
  })

  it('apply class to divider', () => {
    const vm = mount(Divider, {
      attrs: {
        class: 'my-divider',
      },
    })

    expect(vm.classes()).toContain('my-divider')
  })

})
