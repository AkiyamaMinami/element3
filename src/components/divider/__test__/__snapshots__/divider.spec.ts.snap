// Vitest Snapshot v1

exports[`Divider.vue > snapshot 1`] = `
<div
  class="el-divider el-divider--horizontal is-solid"
>
  <!--v-if-->
</div>
`;
